tensorflow/core/common_runtime/eager/kernel_and_device.cc

```diff
@@ -278,6 +280,47 @@ Status KernelAndDeviceOp::Run(
 
   OpKernelContext context(&params);
 
+  string msg;
+  if (nvtx::NvtxRangesEnabled() || nvtx::NvtxRangesDetailedEnabled()) {
```

Comment:
The outer conditional can be refactored away.

Reply:
+1
The whole construction of msg should be in a separate function / library.

Reply:
Applied in d0857f8c16d48d6a28e8be2cc098e6666090ee34